### PR TITLE
Add and define code-style rules in .editorconfig

### PR DESCRIPTION
**Pull request by IgorFlyunt1**
I've initiated the process of integrating and refining our .editorconfig to align with the code-style rules recommended by Microsoft and the broader .NET community. This action aims to enhance consistency and maintainability across our codebase, leveraging best practices from notable .NET repositories like dotnet/runtime, dotnet/aspnetcore, and dotnet/efcore.

**Your Feedback is Crucial:** Before we finalize and enforce these configurations, I highly encourage everyone to review the proposed .editorconfig, and share any feedback or suggestions. Your insights are invaluable in ensuring that our coding standards meet our collective needs and preferences.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,219 @@
+root = true
+
+# default settings:
+[*]
+insert_final_newline = true
+indent_style = space
+indent_size = 2
```
Comment (Marusyk) [inline .editorconfig:7]: Please don't specify an indent_size here; that has too many unintended consequences

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+charset = utf-8
+trim_trailing_whitespace = true
+
+# C# files
+[*.cs]
```
Comment (Marusyk) [inline .editorconfig:12]: Let's also add `max_line_length = 130`

**Review by Marusyk — APPROVED** [review]
Just a small fixes otherwise LGTM 👍
Thanks for your contribution

**Comment by Marusyk** [comment]
would you be interested in setting up GitHub Actions + validation of the editorconfig?

**Review by kanilsz — APPROVED** [review]
I agree to use the recommended Microsoft code style. If anything (un)expected will appear - we could change it on the move.

**Comment by IgorFlyunt1** [comment]
> would you be interested in setting up GitHub Actions + validation of the editorconfig (#12)?

Of course, you can assign this task to me. Also, please keep in mind that I would like to be assigned the issue related to the implementation of MSA.BuildingBlocks.Logging. I have already started planning the implementation."

**Pull request merged**